tools/RAiDER/models/gmao.py

```diff
@@ -59,42 +62,60 @@ def _fetch(self, lats, lons, time, out, Nextra=2):
-        DT = time - T0
+        T0   = dt.datetime(2017, 12, 1, 0, 0, 0)
+        ## round time to nearest third hour
+        time = roundTime(time, 3*60*60)
+        DT   = time - T0
         time_ind = int(DT.total_seconds() / 3600.0 / 3.0)
 
         ml_min = 0
         ml_max = 71
-
-        # open the dataset and pull the data
-        url = 'https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv'
-        session = pydap.cas.urs.setup_session('username', 'password', check_url=url)
-        ds = pydap.client.open_url(url, session=session)
-
-        q = ds['qv'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
-        p = ds['pl'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
-        t = ds['t'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
-        h = ds['h'].array[
-            time_ind,
-            ml_min:(ml_max + 1),
-            lat_min_ind:(lat_max_ind + 1),
-            lon_min_ind:(lon_max_ind + 1)
-        ][0]
+        if time >= T0:
+            # open the dataset and pull the data
+            url = 'https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv'
+            session = pydap.cas.urs.setup_session('username', 'password', check_url=url)
+            ds = pydap.client.open_url(url, session=session)
+            q  = ds['qv'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+
+            p = ds['pl'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+            t = ds['t'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+            h = ds['h'].array[
+                time_ind,
+                ml_min:(ml_max + 1),
+                lat_min_ind:(lat_max_ind + 1),
+                lon_min_ind:(lon_max_ind + 1)
+            ][0]
+
+        else:
+            root = 'https://portal.nccs.nasa.gov/datashare/gmao/geos-fp/das/Y{}/M{:02d}/D{:02d}'
+            base = f'GEOS.fp.asm.inst3_3d_asm_Nv.{time.strftime("%Y%m%d")}_{time.hour:02}00.V01.nc4'
+            url  = f'{root.format(time.year, time.month, time.day)}/{base}'
+            session = requests_retry_session()
+            resp    = session.get(url, stream=True)
+            assert resp.ok, f'Could not access url for time: {time}'
```

Comment:
Could we enable additional reporting on how the download is progressing using requests?
If not possible, can we add a logging statement saying which date we are downloading?